pyof/foundation/basic_types.py

```diff
@@ -114,13 +117,15 @@ class DPID(GenericType):
 
     _fmt = "!8B"
 
-    def __init__(self, dpid=None):
+    def __init__(self, value=None):
```

Comment:
Dpid is already the class name, there is no need to be redundant.
The same applies to the other modifications.

GenericType accepts a ``value`` parameter in its initialization. The value can be retrieved with the ``.value`` attribute. I think it makes sense to keep the same behavior for its derivatives.
Furthermore, this allows ``type(self)(value=x)`` to work as expected for any ``GenericType``.